torch/optim/sgd.py

```diff
@@ -427,7 +427,12 @@ def _multi_tensor_sgd(
                 device_grads = bufs
 
         if not device_has_sparse_grad:
-            torch._foreach_add_(device_params, device_grads, alpha=-lr)
+            # handle internal item() call if lr is a tensor
+            if isinstance(lr, torch.Tensor) and torch._utils.is_compiling():
```

Comment:
Everything else looks good, but I don’t like adding more is_compiling()s into the code.

What is the plan for compile to support scalar tensors like this? Are there alternatives where is_compiling wouldn’t be necessary?

Reply:
Dynamo handles scalar tensors already, the issue is that passing a scalar as an alpha or value kwarg in some torch functions results in an item() call internally. Symints are the long term plan there, but they aren't ready. Unfortunately I don't think there is an alternative here.